lib/posthog/application.ex

```diff
@@ -0,0 +1,16 @@
+defmodule Posthog.Application do
+  @moduledoc false
+
+  use Application
+
+  def start(_type, _args) do
+    children = [
+      # Start Cachex for feature flag event deduplication.
+      # The 50,000 entries limit is the same used for posthog-python, but otherwise arbitrary.
+      {Cachex, name: :posthog_feature_flag_cache, limit: 50_000, policy: Cachex.Policy.LRW}
```

Comment:
Does this make sense? Do we not wanna send `$feature_flag_called` in case we have this in the cache? This lasts across requests - it's a separate process.

Wont this mean that users with less than `50000` clients wont ever have more than 50000 FFs charged to them? (Users can make these supervisors survive servers restarts, Elixir is crazy)

Reply:
Well, it's the combination of `distinct_id` and `feature_flag`. So yeah, if they only have 1 flag and less than 50,000 distinct users, they'd never be charged for more than 50,000.

I took this approach from `posthog-python`. Many of the other clients do something similar. It seems to me that we should put a time limit on each cache entry. Something like one minute. @dmarticus, do you have context on why we do this in `posthog-python`?

Reply:
Ah, the hash includes the `feature_flag` too, so it's a slightly smaller problem.
The problem is that servers probably restart pretty often in Python; that isn't necessarily true in Elixir: you can keep the `Cachex` process running indefinitely, so... it might fill the cache pretty quickly.

Will leave that decision up to y'all, just letting you know the quirkyness of working with a distributed language :)

Reply:
One other thing to consider, we don't charge based on the number of `$feature_flag_called` events, we charge based on the number of `/decide` calls made. So this cache doesn't affect the cost. Just reporting.

Reply:
Ah, I didn't know that. In that case, I'm fine with keeping it this way. I'm working on improving the setup, I don't think the LRU we added actually works how we think, expect changes to this PR through the day.

Reply:
The one thing I worry about with such a big cache is people being confused why they don't see these events. My gut tells me the reason for this is if they're calling a feature flag in a tight loop, we don't want 10 of the same events all of a sudden.

So I'd suggest we add a `default_ttl` to the `Cachex` declaration. Something like 5 minutes would be fine. The 50,000 limit then just ensures we don't ever take up too much memory.